src/core/ext/transport/chttp2/transport/chttp2_transport.h

```diff
@@ -112,10 +112,22 @@ class HttpAnnotation : public CallTracerAnnotationInterface::Annotation {
     kEnd,
   };
 
-  HttpAnnotation(
-      Type type, gpr_timespec time,
-      absl::optional<chttp2::TransportFlowControl::Stats> transport_stats,
-      absl::optional<chttp2::StreamFlowControl::Stats> stream_stats);
+  HttpAnnotation(Type type, gpr_timespec time);
+
+  HttpAnnotation& Add(const chttp2::TransportFlowControl::Stats& stats) {
+    transport_stats_ = stats;
+    return *this;
+  }
+
+  HttpAnnotation& Add(const chttp2::StreamFlowControl::Stats& stats) {
+    stream_stats_ = stats;
+    return *this;
+  }
+
+  HttpAnnotation& Add(size_t target_wire_size) {
```

Comment:
We're going to regret this.... suggest wrapping it in a struct that has a name....

```
struct WriteSizeStats { size_t target_wire_size; }
HttpAnnotation& Add(WriteSizeStats stats);
```

Reply:
Updated.